src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const copy = { ...state };
+
+  for (const action in actions) {
+    if (actions[action].type === 'addProperties') {
+      Object.assign(copy, actions[action].extraData);
+    };
+
+    if (actions[action].type === 'removeProperties') {
+      for (let i = 0; i < actions[action].keysToRemove.length; i++) {
```

Comment:
Можно использовать цикл for-of:
```suggestion
      for (const key of action.keysToRemove) {
```